tool/wizard/wizard.go

```diff
@@ -0,0 +1,147 @@
+	"github.com/aws/amazon-cloudwatch-agent/tool/processors/serialization"
+	"github.com/aws/amazon-cloudwatch-agent/tool/processors/tracesconfig"
+	"github.com/aws/amazon-cloudwatch-agent/tool/runtime"
+	"github.com/aws/amazon-cloudwatch-agent/tool/stdin"
+	"github.com/aws/amazon-cloudwatch-agent/tool/testutil"
+	"github.com/aws/amazon-cloudwatch-agent/tool/util"
+	wizardflags "github.com/aws/amazon-cloudwatch-agent/tool/wizard/flags"
+)
+
+type IMainProcessor interface {
+	VerifyProcessor(_processor interface{})
+}
+
+type MainProcessorStruct struct{}
+
+var MainProcessorGlobal IMainProcessor = &MainProcessorStruct{}
+
+type Params struct {
+	IsNonInteractiveWindowsMigration bool
+	IsNonInteractiveLinuxMigration   bool
+	TracesOnly                       bool
+	UseParameterStore                bool
+	IsNonInteractiveXrayMigration    bool
+	ConfigFilePath                   string
+	ConfigOutputPath                 string
+	ParameterStoreName               string
+	ParameterStoreRegion             string
+}
+
+func init() {
+	stdin.Scanln = func(a ...interface{}) (int, error) {
+		var n int
+		var err error
+
+		scanner := bufio.NewScanner(os.Stdin)
+		scanner.Scan()
+		if len(a) > 0 {
+			*a[0].(*string) = scanner.Text()
+			n = len(*a[0].(*string))
+		}
+		err = scanner.Err()
+		return n, err
+	}
+	processors.StartProcessor = basicInfo.Processor
+}
+
+func RunWizard(params Params) error {
+	if params.IsNonInteractiveWindowsMigration {
+		addWindowsMigrationInputs(
+			params.ConfigFilePath,
+			params.ParameterStoreName,
+			params.ParameterStoreRegion,
+			params.UseParameterStore,
+		)
+	} else if params.IsNonInteractiveLinuxMigration {
+		ctx := new(runtime.Context)
+		config := new(data.Config)
+		ctx.HasExistingLinuxConfig = true
+		ctx.ConfigFilePath = params.ConfigFilePath
+		if ctx.ConfigFilePath == "" {
+			ctx.ConfigFilePath = wizardflags.DefaultFilePathLinuxConfiguration
+		}
+		process(ctx, config, linux.Processor, serialization.Processor)
+		return nil
+	} else if params.TracesOnly {
+		ctx := new(runtime.Context)
+		config := new(data.Config)
+		ctx.TracesOnly = true
+		ctx.ConfigOutputPath = params.ConfigOutputPath
+		ctx.NonInteractiveXrayMigration = params.IsNonInteractiveXrayMigration
+		process(ctx, config, tracesconfig.Processor, serialization.Processor)
+		return nil
+	}
+
+	startProcessing(params.ConfigOutputPath, params.IsNonInteractiveWindowsMigration, params.IsNonInteractiveXrayMigration)
+	return nil
+}
+
+func RunWizardFromFlags(flags map[string]*string) error {
+	params := Params{
+		IsNonInteractiveWindowsMigration: *flags["isNonInteractiveWindowsMigration"] == "true",
+		IsNonInteractiveLinuxMigration:   *flags["isNonInteractiveLinuxMigration"] == "true",
+		TracesOnly:                       *flags["tracesOnly"] == "true",
+		UseParameterStore:                *flags["useParameterStore"] == "true",
+		IsNonInteractiveXrayMigration:    *flags["nonInteractiveXrayMigration"] == "true",
```

Comment:
This means we expect people to call this with
```
--isNonInteractiveLinuxMigration true
```
now instead of
```
--isNonInteractiveLinuxMigration
```
which currently works.